src/esbmc/esbmc_parseoptions.cpp

```diff
@@ -1721,17 +1732,32 @@ bool esbmc_parseoptionst::process_goto_program(
 
       value_set_analysis.update(goto_functions);
     }
-
-    if(cmdline.isset("goto-coverage") || cmdline.isset("make-assert-false"))
+    if(cmdline.isset("add-false-assert"))
     {
       goto_coveraget tmp;
-      tmp.make_asserts_false(goto_functions);
+      tmp.add_false_asserts(goto_functions);
     }
 
-    if(cmdline.isset("goto-coverage") || cmdline.isset("add-false-assert"))
+    //! goto-cov will mutate the asserts added bt esbmc (e.g. overflow-check)
+    if(cmdline.isset("goto-coverage") || cmdline.isset("goto-coverage-claims"))
+    {
+      // for assertion coverage metric
+      options.set_option("make-assert-false", true);
+      // no-simplify, otherwise the coverage will always be 100%
+      options.set_option("no-simplify", true);
+      // for multi-property
+      options.set_option("result-only", true);
+      options.set_option("base-case", true);
+      options.set_option("multi-property", true);
+      options.set_option("keep-verified-claims", true);
+    }
+
+    if(
+      options.get_bool_option("make-assert-false") ||
+      cmdline.isset("make-assert-false"))
```

Comment:
Why do we need to check both, the `options` and the `cmdline`?

Reply:
my mistake